Lib/asyncio/base_events.py

```diff
@@ -1261,10 +1261,22 @@ async def start_tls(self, transport, protocol, sslcontext, *,
             raise TypeError(
                 f'transport {transport!r} is not supported by start_tls()')
 
+        # gh-109051: SSLProtocol needs to preserve "writing paused" state
+        if isinstance(transport, transports._FlowControlMixin):
+            writing_paused = transport._protocol_paused
+        else:
+            # Don't break compatibility with transports that don't implement
+            # the private _FlowControlMixin (e.g. wrapper transports) as much
+            # as possible.
+            _, high_water = transport.get_write_buffer_limits()
+            buffer_size = transport.get_write_buffer_size()
+            writing_paused = buffer_size > high_water
```

Comment:
Would it safe to only use this code path? (remove fast-path for _FlowControlMixin)

Reply:
In case of doubt, keep the specific code path for _FlowControlMixin.